### ISSUE TITLE
Panic when restarting controller

### ISSUE THREAD

**Issue by rbradford** [ciao-project/ciao#1466]
If you restart controller you can sometimes see a panic from the cnci manager causes by a segfault:

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x46a8f2]

goroutine 123 [running]:
sync.(*RWMutex).Lock(0x0)
	/usr/local/go/src/sync/rwmutex.go:93 +0x22
main.transitionInstanceState(0xc4201a8000, 0x9d33f6, 0x6, 0x0, 0x0)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/instance.go:183 +0x5f
main.(*CNCI).transitionState(0xc420213170, 0x9d33f6, 0x6)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/cnci.go:105 +0x11a
main.(*CNCIManager).CNCIAdded(0xc4201e2d40, 0xc420230db0, 0x24, 0x0, 0x0)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/cnci.go:365 +0xda
main.(*ssntpClient).concentratorInstanceAdded(0xc4200a6fc0, 0xc4203fc630, 0xac, 0xac)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/client.go:226 +0x39d
main.(*ssntpClient).EventNotify(0xc4200a6fc0, 0x9f5303, 0xc4201e20c0)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/client.go:325 +0x23b
github.com/ciao-project/ciao/ssntp.(*Client).processSSNTPFrame(0xc4200a6fc8, 0xc4201e20c0)
	/home/rob/go/src/github.com/ciao-project/ciao/ssntp/client.go:101 +0x132
created by github.com/ciao-project/ciao/ssntp.(*Client).handleSSNTPServer
	/home/rob/go/src/github.com/ciao-project/ciao/ssntp/client.go:142 +0xa9
```

**Comment by rbradford** [comment]
Slightly different panic but again caused by trying to start an instance after controller restart (but the CNCI is still running.)

```
2017/09/28 16:41:15 http: panic serving 127.0.0.1:46206: runtime error: invalid memory address or nil pointer dereference
goroutine 212 [running]:
net/http.(*conn).serve.func1(0xc42008b180)
	/home/rob/opt/go/src/net/http/server.go:1697 +0xd0
panic(0x94afa0, 0xe4eff0)
	/home/rob/opt/go/src/runtime/panic.go:491 +0x283
sync.(*RWMutex).RLock(0x0)
	/home/rob/opt/go/src/sync/rwmutex.go:48 +0x27
main.instanceActive(0xc42019c000, 0xc420209200)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/instance.go:208 +0x3b
main.(*CNCIManager).waitForActive(0xc4201dce80, 0x1000, 0x1000, 0xc4201ca4c8)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/cnci.go:400 +0xa4
main.(*CNCIManager).WaitForActive(0xc4201dce80, 0x1000, 0x0, 0x0)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/cnci.go:196 +0xdf
github.com/ciao-project/ciao/ciao-controller/internal/datastore.(*Datastore).AllocateTenantIP(0xc4200b0b00, 0xc420020ea0, 0x24, 0x24, 0x7aa751, 0xc4200143e0, 0xc4201881b0, 0xc420472f8b)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/internal/datastore/datastore.go:750 +0x86a
main.networkConfig(0xc4200f8af0, 0xc420110a00, 0xc420422290, 0xc420110a00, 0x0, 0x0)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/instance.go:345 +0x149
main.newConfig(0xc4200f8af0, 0xc420422d30, 0xc420252780, 0x24, 0xc420472f8b, 0x24, 0x0, 0x0, 0x0, 0x0, ...)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/instance.go:435 +0x196
main.newInstance(0xc4200f8af0, 0xc420472f8b, 0x24, 0xc420422d30, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/instance.go:81 +0x114
main.(*controller).startWorkload(0xc4200f8af0, 0xc420252750, 0x24, 0xc420472f8b, 0x24, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/command.go:237 +0x28a
main.(*controller).CreateServer(0xc4200f8af0, 0xc420472f8b, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc420252750, ...)
	/home/rob/go/src/github.com/ciao-project/ciao/ciao-controller/openstack_compute.go:377 +0x257
github.com/ciao-project/ciao/openstack/compute.createServer(0xc420203940, 0xe61fe0, 0xc42010cb60, 0xc4203a5f00, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/rob/go/src/github.com/ciao-project/ciao/openstack/compute/api.go:528 +0x23d
github.com/ciao-project/ciao/openstack/compute.APIHandler.ServeHTTP(0xc420203940, 0x9f4708, 0xe61fe0, 0xc42010cb60, 0xc4203a5f00)
	/home/rob/go/src/github.com/ciao-project/ciao/openstack/compute/api.go:474 +0x72

```

**Comment by rbradford** [comment]
Fixed by #1473
